### PR TITLE
Allow elasticsearch to be configurable

### PR DESCRIPTION
**Pull request by bderusha**
ToDo:
- [x] Update tests

### CODE REVIEW

File: search/elastic.py [hunk 1]
```diff
@@ -270,7 +270,8 @@ def _clear_mapping(self, doc_type):
 
     def __init__(self, index=None):
         super(ElasticSearchEngine, self).__init__(index)
-        self._es = getattr(settings, "ELASTIC_SEARCH_IMPL", Elasticsearch)()
+        es_config = getattr(settings, "ELASTIC_SEARCH_CONFIG", [{}])
```
Comment (dianakhuang) [inline search/elastic.py]: I'm a little confused as to why this takes in a list of hosts. Is that just for fallbacks?

Comment (bderusha) [inline search/elastic.py]: I believe so.  But it always works with host configs as a list in case you need more than one.

**Comment by dianakhuang** [comment]
Would it or could it be possible to make the port configurable as well? I don't know if our production version of ElasticSearch uses the same port as the one that we've been using by default.

**Comment by bderusha** [comment]
@dianakhuang the port is configurable.  Elasticsearch is pretty smart about the config.

http://elasticsearch-py.readthedocs.org/en/master/api.html#elasticsearch.Elasticsearch

```python
es = Elasticsearch([
    {'host': 'localhost'},
    {'host': 'othernode', 'port': 443, 'url_prefix': 'es', 'use_ssl': True},
])
```

```python
es = Elasticsearch(
    ['localhost:443', 'other_host:443'],
    # turn on SSL
    use_ssl=True,
    # make sure we verify SSL certificates (off by default)
    verify_certs=True,
    # provide a path to CA certs on disk
    ca_certs='/path/to/CA_certs'
)
```

**Comment by dianakhuang** [comment]
@bderusha Ah, okay. Excellent! :rocket:

**Comment by bderusha** [comment]
@andy-armstrong could you take a look at this as well?

**Comment by andy-armstrong** [comment]
:+1:

**Pull request merged**